sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/FnApiStateAccessor.java

```diff
@@ -128,7 +128,7 @@ public FnApiStateAccessor(
       @Override
       public ResultT get() {
         ArgT currentArg = arg.get();
-        if (currentArg != memoizedArg) {
+        if (currentArg != memoizedArg || currentArg == null) {
```

Comment:
Why do you need to check if it is null?

Reply:
For VoidCoder the arg.get() returns null, which means the currentArg != memoizedArg will evaluate to false. 
This leaves the memoizedResult to null(underneath function never applied), and for VoidCoder key coder we need to set the key to ByteString.EMPTY instead of null(generated proto class doesn't allow setting to null) I believe.

Reply:
I realized this and updated my comment just after posting to use a boolean.